### PR TITLE
minor: Update GitHub label workflow.

### LINKED ISSUE #2617: Update GitHub label workflow
**Is your feature request related to a problem or challenge? Please describe what you are trying to do.**
I would like to propose some changes to the way that labels are automatically applied.

I don't think it is helpful now to have a `datafusion` label now that Ballista has moved

I would like labels to be added based on the crate and/or module being updated. For example:

- `logical-expr` label for changes to `datafusion-expr` crate
- `physical-expr` label for changes to `datafusion-physical-expr` crate
- `sql` label for changes to `datafusion-sql` crate
- `optimizer` label for changes to `datafusion/src/optimizer` module
- `core` for any other changes in the core `datafusion` crate

**Describe the solution you'd like**
As above

**Describe alternatives you've considered**
None

**Additional context**
None

### PR DESCRIPTION
**Pull request by jackwener**
# Which issue does this PR close?

Closes #2617 .

 # Rationale for this change

 Update GitHub label workflow.

# What changes are included in this PR?

# Are there any user-facing changes?
None

# Does this PR break compatibility with Ballista?
None

### CODE REVIEW

**Comment by jackwener** [comment]
What'more, the label need to be added by maintainer.

File: .github/workflows/dev_pr/labeler.yml [hunk 1]
```diff
@@ -38,3 +38,16 @@ documentation:
   - ./**/README.md
   - DEVELOPERS.md
   - datafusion/docs/**.*
+
```
Comment (andygrove) [inline .github/workflows/dev_pr/labeler.yml:41]:
```suggestion

sql:
  - datafusion/sql/**/*
  
```

**Comment by andygrove** [comment]
Thanks @jackwener! I added a suggestion add `sql` as well. I will create that labels

**Review by Ted-Jiang — COMMENTED** [review]
https://github.com/apache/arrow-datafusion/blob/907504c5aa768601f9d70ad2c8f928bedfa9b069/.github/workflows/dev_pr/labeler.yml#L26-L27
Not exist, maybe we can delete in this pr.

**Review by andygrove — APPROVED** [review]

**Review by andygrove — APPROVED** [review]
Thanks again @jackwener

**Pull request merged**